### ISSUE TITLE
Material list "Upgrade"

### ISSUE THREAD

**Issue by Quits123** [maruohon/litematica#779]
I am always annoyed when I have a big project and then all the items don't fit in my inventory. I think a feature that could be added is you select an area with your chests/shulkers/ any other storage and the select what youre building it then tells you if you have all the materials you need in the chests or if you need more. Like the material list.

**Comment by maruohon** [comment]
Yes a "prep chests" feature has been on the todo list for the material list for ages. I just haven't gotten to it yet due to the on-going mod rewrites and other stuff.

**Comment by Quits123** [comment]
Ok thank you, this would be great if it could be added as I have a project with over 25,000 blocks and it is kind of complicated to calculate it all

**Comment by mms0316** [comment]
@Quits123, what version of Minecraft are you on? I do have a fork with such a feature, but not for all mc versions.

https://www.youtube.com/watch?v=FRKWOFApqs0
https://www.youtube.com/watch?v=exOqkw2GVQI

**Comment by Quits123** [comment]
1.20-1.20.1

**Comment by Quits123** [comment]
@mms0316 nice video

**Comment by mms0316** [comment]
https://github.com/mms0316/litematica/releases/tag/v1.20.1-0.15.3.mms.36
You can get the .jar from this, or compile it from 237035f

To fetch items that match an opened container and what's in the Material List's Info HUD, use a key that was previously set in Hotkeys -> materialListFetch

To register/unregister a container for the outlines, set a key for Hotkeys -> materialListContainerRegister and materialListContainerUnregister

**Comment by Quits123** [comment]
thanks

**Comment by Scubakay** [comment]
@Quits123 I made an addon mod for Litematica that also checks your enderchest for stuff from the material list:

https://modrinth.com/mod/litematica-enderchest-materials

**Comment by placTc** [comment]
bumping this thread

**Comment by Tenben-FPV** [comment]
Is this now in the Base mod?

**Comment by LiamKrenn** [comment]
bump

**Comment by MBeggiato** [comment]
Any update on this?
